source/common/grpc/async_client_impl.cc

```diff
@@ -60,28 +60,35 @@ AsyncStreamImpl::AsyncStreamImpl(AsyncClientImpl& parent,
     : parent_(parent), service_method_(service_method), callbacks_(callbacks), timeout_(timeout) {}
 
 void AsyncStreamImpl::initialize(bool buffer_body_for_retry) {
-  auto& http_async_client = parent_.cm_.httpAsyncClientForCluster(parent_.remote_cluster_name_);
-  dispatcher_ = &http_async_client.dispatcher();
-  stream_ = http_async_client.start(*this, absl::optional<std::chrono::milliseconds>(timeout_),
-                                    buffer_body_for_retry);
+  try {
```

Comment:
Use cm_.get(cluster_name) to check if cluster exists, instead of long try and blindly catching all `EnvoyException`s

Reply:
+1 I would avoid the try/catch.

Reply:
yes. good points. Will do that.